src/game_api/rpc.cpp

```diff
@@ -943,3 +943,91 @@ void pick_up(uint32_t who, uint32_t what)
         pick_up_func(ent, item);
     }
 }
+
+void set_olmec_phase_y_level(uint8_t phase, float y)
+{
+    // Sets the Y-level Olmec changes phases. The defaults are :
+    // - phase 1 (bombs) = 99 (+1)  (the game adds 1 to the fixed value for some reason)
+    // - phase 2 (ufos) = 83
+    // Olmecs checks phases in order! The means if you want ufo's from the start
+    // you have to put both phase 1 and 2 at e.g. level 199
+    // If you want to make Olmec stay in phase 0 (stomping) all the time, you can just set
+    // the phase 1 y level to e.g. 10.
+    static size_t phase1_offset = 0;
+    static size_t phase2_offset = 0;
+    if (phase1_offset == 0)
+    {
+        auto state = State::get();
+        auto memory = Memory::get();
+        auto exe = memory.exe();
+
+        std::string pattern_phase1 = "\xF3\x0F\x10\x15\x07\x3D\x36\x00"s;
+        std::string pattern_phase2 = "\xF3\x0F\x10\x0D\xCD\x38\x36\x00"s;
+
+        // first look up these patterns so we are in the correct function
+        auto offset1 = find_inst(exe, pattern_phase1, memory.after_bundle);

```

Comment:
Should these be stored somewhere statically?

Reply:
These are intermediary/temporary offsets, and serve only to calculate phase1_offset and phase2_offset, which are static and permanent. The whole pattern detection block is only executed once (if the pattern is found of course) and there is no further use for offset1 and 2, so they don't need to be stored permanently.

Reply:
Oh yes, I didn't read the start of this block properly. It's not looking how I'm used to from OL so I just missed it.